### ISSUE TITLE
Error using tabular/join

### ISSUE THREAD

**Issue by Araenn** [EricMagalhaesDelgado/m2mlapp#3]
I get this error :
Error using tabular/join
The key variable for the right table must contain all values in the key variable for the left table.

Error in util.uiPropertiesParser (line 92)
    uiProperties = join(propNames, propCode);

Error in m2mlapp (line 98)
        uiProperties = util.uiPropertiesParser(matlabCode, treeCode);
        
But I can't figure why. I wrote my "app" in a .m, in Matlab R2023b

**Comment by EricMagalhaesDelgado** [comment]
Hi Araenn!
Sorry it took me a while to get back to this issue. Listen... if you're still facing the problem and are able to share the file, I’d be happy to try and help!

**Comment by Achuan-2** [comment]
[CalciumDeltaFCaculator.zip](https://github.com/user-attachments/files/20261993/CalciumDeltaFCaculator.zip)

I met the same problem
@EricMagalhaesDelgado

**Comment by Achuan-2** [comment]
I find it out that the order of the Properties declaration of app components and the createComponents(app) declaration must be consistent, otherwise an error will occur

**Comment by EricMagalhaesDelgado** [comment]
Ok! There was a comment in the component declarations, and I wasn’t expecting that. I’ve fixed the bug — that was the issue. You can run it again now. Thanks for the feedback!

**Comment by Achuan-2** [comment]
> Ok! There was a comment in the component declarations, and I wasn’t expecting that. I’ve fixed the bug — that was the issue. You can run it again now. Thanks for the feedback!

thanks a lot, thank you for your code, makes it easier to use AI to modify the code in app.

**Comment by EricMagalhaesDelgado** [comment]
Are you writing MATLAB code outside MATLAB env? VSCode+Copilot?

**Comment by Achuan-2** [comment]
> Are you writing MATLAB code outside MATLAB env? VSCode+Copilot?

yes, I use vscode+copilot, grok, gemini 2.5, claude 3.5/3.7. Sometimes, for convenience, I just let AI write an app from scratch. Like CalciumDeltaFCaculator.m, it's totally written by AI. Converting to mlapp would make it easier for me to tweak the interface and add features.

**Comment by EricMagalhaesDelgado** [comment]
Amazing! So... I noticed a few issues with the way it was written. And yeah... AppDesigner will definitely help you create a better experience for the app's users. I had some free time, so I made a few edits to your file. 🙂

Thinks that you have to work on:
- Use uigridlayout whenever possible.
- Initialize components with default parameters rather than customizing them and then changing them later in the startup function.
- Create uiaxes programmatically and place them over a tiledlayout — this improves behavior, especially when working with real-time plots.

[CalciumDeltaFCaculator.zip](https://github.com/user-attachments/files/20270471/CalciumDeltaFCaculator.zip)

**Comment by EricMagalhaesDelgado** [comment]
I am gonna close this issue right now. See you around!

**Comment by Achuan-2** [comment]
Thank you for your kind suggestion. I indeed rarely use uigridlayout myself, and I will find time to learn it later
